compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -923,7 +923,51 @@ trait Implicits:
           // example where searching for a nested type causes an infinite loop.
           None
 
-    MissingImplicitArgument(arg, pt, where, paramSymWithMethodCallTree, ignoredInstanceNormalImport)
+    def allImplicits(currImplicits: ContextualImplicits): List[ImplicitRef] =
+      if currImplicits.outerImplicits == null then currImplicits.refs
+      else currImplicits.refs ::: allImplicits(currImplicits.outerImplicits)
+
+    /** Ensure an implicit is not a Scala 2-style implicit conversion, based on its type */
+    def isScala2Conv(typ: Type): Boolean = typ match {
```

Comment:
Maybe something like `isImplicitDefConversion` would be more accurate? In scala 2 also implicit instances of subtypes of `Function1` were treated as conversions but we don't take them into account here